pkg/operator/ceph/cluster/controller.go

```diff
@@ -496,23 +497,50 @@ func (c *ClusterController) checkPVPresentInCluster(drivers []string, clusterID
 	return false, nil
 }
 
-// removeFinalizer removes a finalizer
-func removeFinalizer(client client.Client, name types.NamespacedName) error {
-	cephCluster := &cephv1.CephCluster{}
-	err := client.Get(context.TODO(), name, cephCluster)
+func removeFinalizers(client client.Client, name types.NamespacedName) error {
+	// Remove cephcluster finalizer
+	err := removeFinalizer(client, name, &cephv1.CephCluster{}, "")
+	if err != nil {
+		return errors.Wrap(err, "failed to remove cephcluster finalizer")
+	}
+
+	// Remove finalizer for rook-ceph-mon secret
+	name = types.NamespacedName{Name: mon.AppName, Namespace: name.Namespace}
+	err = removeFinalizer(client, name, &corev1.Secret{}, mon.DisasterProtectionFinalizerName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to remove finalizer for the secret %q", name.Name)
+	}
+
+	// Remove finalizer for rook-ceph-mon-endpoints configmap
+	name = types.NamespacedName{Name: mon.EndpointConfigMapName, Namespace: name.Namespace}
+	err = removeFinalizer(client, name, &corev1.ConfigMap{}, mon.DisasterProtectionFinalizerName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to remove finalizer for the configmap %q", name.Name)
+	}
+	return nil
+}
+
+func removeFinalizer(client client.Client, name types.NamespacedName, obj client.Object, finalizer string) error {
+	err := client.Get(context.TODO(), name, obj)
 	if err != nil {
 		if kerrors.IsNotFound(err) {
-			logger.Debug("CephCluster resource not found. Ignoring since object must be deleted.")
+			logger.Debugf("%s resource not found. Ignoring since object must be deleted.", name.Name)
 			return nil
 		}
-		return errors.Wrapf(err, "failed to retrieve ceph cluster %q to remove finalizer", name.Name)
+		return errors.Wrapf(err, "failed to retrieve %q to remove finalizer", name.Name)
 	}
 
-	err = opcontroller.RemoveFinalizer(client, cephCluster)
+	if finalizer == "" {
+		err = opcontroller.RemoveFinalizer(client, obj)
```

Comment:
The `RemoveFinalizer` method called here also invokes the K8s API to make a `Get` request. So we are making the request twice, first in line 524 and then inside `RemoveFinalizer` method in line 534

Reply:
https://github.com/rook/rook/blob/50fb1b708684a5876dd97536916857c4cb619b47/pkg/operator/ceph/controller/finalizer.go#L78

We can remove this call if you say, it is recently added.

Reply:
Ignore my comment. Lets not remove it.  This method is used in other places as well. Looks like we are trying to get the latest version of the object. So it should be ok.